csharp/src/Google.Protobuf/Reflection/Dynamic/DynamicMessage.cs

```diff
@@ -0,0 +1,387 @@
+﻿using Google.Protobuf.Collections;
+using System;
+using System.Collections;
+
+namespace Google.Protobuf.Reflection.Dynamic
```

Comment:
I think we'll be able to use just Google.Protobuf.Reflection for this. I don't think a new sub-namespace is required.